### PR TITLE
Amplitude tracking: SignUp, UserMessagePosted, WorkspaceGroups

### PR DESCRIPTION
**Pull request by lasryaric**
## Description

Adding  Amplitude tracking for the first events:
- SignUp
- UserMessagePosted

This also contains association between users and workspaces.

### CODE REVIEW

File: front/lib/amplitude/back/generated/index.ts [hunk 1]
```diff
@@ -5,16 +5,16 @@
  * Ampli - A strong typed wrapper for your Analytics
  *
```
Comment (lasryaric) [inline front/lib/amplitude/back/generated/index.ts:6]: Generated code, no need to look at it.

File: front/lib/amplitude/browser/generated/index.ts [hunk 2]
```diff
@@ -5,35 +5,35 @@
  * Ampli - A strong typed wrapper for your Analytics
```
Comment (lasryaric) [inline front/lib/amplitude/browser/generated/index.ts:5]: Generated code, no need to look at it.

File: front/lib/amplitude/back/index.ts [hunk 3]
```diff
@@ -12,10 +17,71 @@ export function getBackendClient() {
...
+  userId,
+  conversationId,
+}: {
+  userMessage: UserMessageType;
+  workspace: WorkspaceType;
+  userId: string;
+  conversationId: string;
+}) {
+  const amplitude = getBackendClient();
+  amplitude.userMessagePosted(
```
Comment (flvndvd) [inline front/lib/amplitude/back/index.ts:71]: Unawaited promise?

File: front/lib/amplitude/back/index.ts [hunk 4]
```diff
@@ -12,10 +17,71 @@ export function getBackendClient() {
...
+    });
+    amplitude.client.setGroup("Group Name", workspace.name, {
+      user_id: `user-${user.id}`,
+      time: membership.createdAt.getTime(),
+      insert_id: `membership_${user.id}-${workspace.name}`,
+    });
+  }
+}
+
+export async function trackUser(user: UserType) {
```
Comment (flvndvd) [inline front/lib/amplitude/back/index.ts:50]: Does it need to be async?

File: front/lib/amplitude/config.ts [hunk 5]
```diff
@@ -0,0 +1,2 @@
+// This is a public API key, which will endup in the user's browser anyway.
+export const AMPLITUDE_PUBLIC_API_KEY = "6ba33096c77a939358f9c21e12d73592";
```
Comment (flvndvd) [inline front/lib/amplitude/config.ts:2]: All the tracked events are stored in the back-end. Can you shed some light on why it's necessary to have this specific key in the front-end?

File: front/lib/amplitude/back/index.ts (hunk 3 above)
Comment (lasryaric) [reply front/lib/amplitude/back/index.ts:71]: We are not waiting for amplitude promise, they are wrapped in an object anyway, and the logger will let us know if something went wrong. This function should not be async.

File: front/lib/amplitude/config.ts (hunk 5 above)
Comment (lasryaric) [reply front/lib/amplitude/config.ts:2]: There is a PR in review for the quick guide, which will use Amplitude on the front-end side to measure how much time people are spending on the guide.

**Review by lasryaric — COMMENTED** [review]
Answered.

**Pull request merged**